definition/rest/v1/settings.ts

```diff
@@ -10,38 +10,38 @@ type SettingsUpdatePropsActions = {
 export type OauthCustomConfiguration = {
 	_id: string;
 	clientId?: string;
-	custom: unknown;
+	custom: boolean;
 	service?: string;
-	serverURL: unknown;
-	tokenPath: unknown;
-	identityPath: unknown;
-	authorizePath: unknown;
-	scope: unknown;
+	serverURL: string;
+	tokenPath: string;
+	identityPath: string;
+	authorizePath: string;
+	scope: string;
 	loginStyle: 'popup' | 'redirect';
-	tokenSentVia: unknown;
-	identityTokenSentVia: unknown;
-	keyField: unknown;
-	usernameField: unknown;
-	emailField: unknown;
-	nameField: unknown;
-	avatarField: unknown;
-	rolesClaim: unknown;
-	groupsClaim: unknown;
-	mapChannels: unknown;
-	channelsMap: unknown;
-	channelsAdmin: unknown;
-	mergeUsers: unknown;
-	mergeRoles: unknown;
-	accessTokenParam: unknown;
-	showButton: unknown;
-
-	appId: unknown;
+	tokenSentVia: string | boolean;
+	identityTokenSentVia: string | boolean;
+	keyField: string | boolean;
+	usernameField: string | boolean;
+	emailField: string | boolean;
+	nameField: string | boolean;
+	avatarField: string | boolean;
+	rolesClaim: string;
+	groupsClaim: string;
+	mapChannels: string;
+	channelsMap: string;
+	channelsAdmin: string;
+	mergeUsers: boolean;
+	mergeRoles: boolean;
+	accessTokenParam: string;
+	showButton: boolean;
+
+	appId: string;
 	consumerKey?: string;
 
-	clientConfig: unknown;
-	buttonLabelText: unknown;
-	buttonLabelColor: unknown;
-	buttonColor: unknown;
+	clientConfig: { provider: SettingValue };
```

Comment:
as you can see below, `clientConfig` can have different values, so maybe let this as `unknown`:

https://github.com/RocketChat/Rocket.Chat/blob/d42f1643f1b6ceff55ae079509879a4433741ffb/app/lib/server/startup/oAuthServicesUpdate.js#L100